lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -1070,6 +1127,122 @@ def plot(self, *, show_outline=True, show_labels=True, axes=None, **kwargs):
+            Kymograph frame
+
+        Returns
+        -------
+        list of tuple of int
+            Returns a list of tuples. This list has one entry for each track that intersects with
+            this kymograph frame. The tuple consists of the track index and the index of which
+            coordinate inside that track corresponds to this kymograph frame.
+
+        Raises
+        ------
+        NotImplementedError
+            If this group has multiple kymographs associated with it.
+        """
+        if not self:
+            return []  # _validate_single_source throws for empty groups
+
+        self._validate_single_source("_tracks_in_frame")
+
+        tracks_in_frame = []
+        for track_idx, track in enumerate(self):
+            track_frame_idx = np.where(track.time_idx == frame_idx)[0]
+            if track_frame_idx.size > 0:
+                tracks_in_frame.append((track_idx, track_frame_idx))
+
+        return tracks_in_frame
+
+    def plot_fit(self, frame_idx, *, fit_kwargs=None, data_kwargs=None, show_track_index=True):
+        """Plot the localization model fits for a particular kymograph frame.
+
+        Note that currently, this fit is only available for Gaussian refinement.
+
+        Parameters
+        ----------
+        frame_idx : int
+            Index of the kymograph frame to plot.
+        fit_kwargs : dict
+            Dictionary containing arguments to be passed to the fit plot.
+        data_kwargs : dict
+            Dictionary containing arguments to be passed to the data plot.
+        show_track_index : bool
+            Display track index in the plot.
+
+        Raises
+        ------
+        ValueError
+            If the group is empty.
+        NotImplementedError
+            If this group has multiple kymographs associated with it.
+        NotImplementedError
+            If localization was not performed using Gaussian localization.
+        IndexError
+            If requesting a fit for a frame index out of bounds of the KymoTrack.
+
+        Examples
+        --------
+        ::
+
+            import lumicks.pylake as lk
+
+            tracks = lk.track_greedy(kymo, channel="red", pixel_threshold=5)
+            refined = lk.refine_tracks_gaussian(tracks, window=10, refine_missing_frames=True, overlap_strategy="multiple")
+
+            # Plot the localization fits for kymograph frame 5
+            refined.plot_fit(5)
+
+            # Using ipywidgets and jupyter, it is easy to make an interactive version of this plot.
+            import ipywidgets as widgets
+
+            plt.figure()
+            def plot_fit(frame):
+                plt.cla()  # Clear the current axis
+                refined.plot_fit(frame, show_track_index=True)
+
+            widgets.interact(plot_fit, frame=widgets.IntSlider(0, max=kymo.shape[1] - 1));
+        """
+        self._validate_single_source("plot_fit")
+        if not self._kymos:
+            raise ValueError("No tracks in the group")
```

Comment:
elsewhere when we test for empty group and raise we have a different error message (which is leftover from before we added multi-source functionality so there wouldn't be a breaking change). Might be good to keep it consistent.

In `_validate_single_source` we use
```
if not self:
    raise RuntimeError("No kymo associated with this empty group (no tracks available)")
```
however, in `_calibration_info` we have the same message but raise an `IndexError` 😖

Reply:
Actually, I suspect this isn't needed anymore, since _validate_single_source handles that now (when I first made this branch, that wasn't checked there yet).

That does mean this particular exception wasn't tested though. Added a test case now.